src/PIL/TiffImagePlugin.py

```diff
@@ -1258,94 +1241,87 @@ def _setup(self):
         # build tile descriptors
         x = y = layer = 0
         self.tile = []
-        self.use_load_libtiff = False
-        if STRIPOFFSETS in self.tag_v2:
+        self.use_load_libtiff = READ_LIBTIFF or self._compression != 'raw'
+        if self.use_load_libtiff:
+            # Decoder expects entire file as one tile.
+            # There's a buffer size limit in load (64k)
+            # so large g4 images will fail if we use that
+            # function.
+            #
+            # Setup the one tile for the whole image, then
+            # use the _load_libtiff function.
+
+            # libtiff handles the fillmode for us, so 1;IR should
+            # actually be 1;I. Including the R double reverses the
+            # bits, so stripes of the image are reversed.  See
+            # https://github.com/python-pillow/Pillow/issues/279
+            if fillorder == 2:
+                key = (
+                    self.tag_v2.prefix, photo, sampleFormat, 1,
+                    self.tag_v2.get(BITSPERSAMPLE, (1,)),
```

Comment:
Are all other values except `fillorder` always should be the same as in the previous key? If so, it's better to modify existing key.

Option 1:

```python
key = key[:3] + (1,) + key[4:]
```

Option 2:

```python
key = list(key)
key[3] = 1
key = tuple(key)
```